crates/turbo-tasks-fs/src/glob.rs

```diff
@@ -347,28 +340,25 @@ impl<'a> Iterator for GlobPartMatchesIterator<'a> {
                     None
                 }
             }
-            GlobPart::FileChar(chars) => loop {
-                let end = match self.cursor.next_boundary(self.path, 0) {
-                    Ok(Some(end)) => end,
-                    _ => return None,
+            GlobPart::FileChar(chars) => {
+                let start = self.cursor.cur_cursor();
+                let Ok(Some(end)) = self.cursor.next_boundary(self.path, 0) else {
+                    return None;
                 };
-
-                let c = &chars[self.cursor.cur_cursor()..end]
```

Comment:
This is unused, but it doesn't really make sense to use the path cursor to index `chars`